$MirOS$

	* Fix strcpy/strcat
	* Include licence in binary, for simplicity

--- main.c.orig	Thu Mar 16 17:44:23 2017
+++ main.c	Thu Jul 20 21:49:59 2017
@@ -307,8 +307,9 @@ static void set_c_outname( const char * 
   {
   output_filename = resize_buffer( output_filename, strlen( name ) +
                                    strlen( known_extensions[0].from ) + 1 );
-  strcpy( output_filename, name );
-  strcat( output_filename, known_extensions[0].from );
+  memcpy( output_filename, name, strlen( name ) );
+  memcpy( output_filename + strlen( name ), known_extensions[0].from,
+                                   strlen( known_extensions[0].from ) + 1 );
   }
 
 
@@ -323,14 +324,15 @@ static void set_d_outname( const char * 
       {
       output_filename = resize_buffer( output_filename, name_len +
                                        strlen( known_extensions[eindex].to ) + 1 );
-      strcpy( output_filename, name );
-      strcpy( output_filename + name_len - from_len, known_extensions[eindex].to );
+      memcpy( output_filename, name, name_len );
+      memcpy( output_filename + name_len - from_len, known_extensions[eindex].to,
+                                       strlen( known_extensions[eindex].to ) + 1 );
       return;
       }
     }
   output_filename = resize_buffer( output_filename, name_len + 4 + 1 );
-  strcpy( output_filename, name );
-  strcat( output_filename, ".out" );
+  strlcpy( output_filename, name, name_len + 4 + 1 );
+  strlcat( output_filename, ".out", name_len + 4 + 1 );
   if( verbosity >= 1 )
     fprintf( stderr, "%s: Can't guess original name for '%s' -- using '%s'\n",
              program_name, name, output_filename );
@@ -1014,7 +1016,8 @@ int main( const int argc, const char * c
             {
             output_filename = resize_buffer( output_filename,
                                              strlen( default_output_filename ) + 1 );
-            strcpy( output_filename, default_output_filename );
+            memcpy( output_filename, default_output_filename,
+                                             strlen( default_output_filename ) + 1 );
             }
           if( !open_outstream( force, true ) )
             {
@@ -1084,3 +1087,6 @@ int main( const int argc, const char * c
   ap_free( &parser );
   return retval;
   }
+
+__asm__(".section .rodata"
+    "\n	.incbin	\"COPYING\"");
